herd/AArch64ASLSem.ml

```diff
@@ -1049,6 +1061,9 @@ module Make (TopConf : AArch64Sig.Config) (V : Value.AArch64ASL) :
       let tr_cnstrnt acc = function
         | ASLVC.Warn s -> M.VC.Warn s :: acc
         | ASLVC.Failed e -> M.VC.Failed e :: acc
+        | ASLVC.Assign (ASLS.A.V.Val (Constant.Concrete (ASLScalar.S_Bool true)), ASLVC.Binop (Op.Eq, e1, e2))
+            ->
+            M.VC.Assign (tr_v e1, M.VC.Atom (tr_v e2)) :: acc
```

Comment:
Why so?

Reply:
With the optimisation, `CSET02` has one additional final state. If I delete the lines above, the extra final state disappears. I do not know how this is possible...